packages/room/peer/peer.js

```diff
@@ -451,171 +425,210 @@ export const createPeer = ({ api, createStream, event, streams, config }) => {
+        } else {
+          // TODO: Set codec preferences by modifying the SDP
+          for (const audioCodec of preferredAudioCodecs) {
+            this._pendingPreferredCodecs.audio.push(audioCodec)
+          }
         }
 
         audioTrack.addEventListener('ended', () => {
-          if (!this._peerConnection || !audioTsvr.sender) return
-          this._peerConnection.removeTrack(audioTsvr.sender)
+          if (!this._peerConnection || !audioTransceiver.sender) return
+          this._peerConnection.removeTrack(audioTransceiver.sender)
           this.removeStream(stream.id)
         })
       }
 
+      if (stream.source === 'media') {
+        this._addVideoTransceiver(stream, config, 'webcam')
+      } else if (stream.source === 'screen') {
+        this._addVideoTransceiver(stream, config, 'screen')
+      }
+    }
+
+    /**
+     * @param {import('../stream/stream-types.js').RoomStreamType.InstanceStream} stream
+     * @param {import('../room-types.js').RoomType.Config} config
+     * @param {'webcam' | 'screen'} type
+     */
+    _addVideoTransceiver = (stream, config, type) => {
+      if (!this._peerConnection) return
+
+      const supportsSetCodecPreferences =
+        window.RTCRtpTransceiver &&
+        'setCodecPreferences' in window.RTCRtpTransceiver.prototype
+      const browserName = getBrowserName() || ''
       /** @type {MediaStreamTrack | undefined} */
       const videoTrack = stream.mediaStream.getVideoTracks()[0]
+      /** @type {RTCRtpCodecCapability[]} */
+      const preferredCodecs = []
+
+      if (videoTrack) {
+        const systemVideoCodecs =
+          RTCRtpSender.getCapabilities('video')?.codecs || []
+
+        for (const videoCodec of config.media[type].videoCodecs) {
+          for (const systemVideoCodec of systemVideoCodecs) {
+            if (
+              systemVideoCodec.mimeType.toLowerCase() ===
+              videoCodec.toLowerCase()
+            ) {
+              preferredCodecs.push(systemVideoCodec)
+              systemVideoCodecs.splice(
+                systemVideoCodecs.indexOf(systemVideoCodec),
+                1
+              )
+            }
+          }
+        }
 
-      const browserName = getBrowserName()
-      const simulcastBrowsers = [SAFARI, CHROME, EDGE, OPERA]
+        // Add all remaining codecs
+        for (const videoCodec of systemVideoCodecs) {
+          preferredCodecs.push(videoCodec)
+        }
 
-      let svc = false
+        /** @type {RTCRtpTransceiverInit} */
+        const transceiverInit = {
+          direction: 'sendonly',
+          streams: [stream.mediaStream],
+        }
 
-      /** @type {RTCRtpCodecCapability[]} */
-      const videoPreferedCodecs = []
-      const videoCodecs = RTCRtpReceiver.getCapabilities('video')?.codecs
+        const svcEnabled =
+          config.media[type].svc &&
+          config.media[type].scalabilityMode &&
+          config.media[type].simulcast &&
```

Comment:
Is this means svc only enabled if simulcast is set even simulcast config is false? Why we need to have all config check here? Is it to make sure the config is not empty?

Reply:
The code was updated. `svcEnabled` is set to true when the svc config is true and the browser is not Firefox.